source/adapters/level_zero/command_buffer.cpp

```diff
@@ -22,6 +22,71 @@
 
 namespace {
 
+// Checks whether zeCommandListImmediateAppendCommandListsExp can be used for a
+// given Context and Device.
+bool checkImmediateAppendSupport(ur_context_handle_t Context,
```

Comment:
longer-term, is there anything stopping us from deprecating the old non-immediate path? I don't like how the two modes are intermingling together. It makes the code more complex.

Reply:
One consideration is the supported driver versions, we need to make sure this works with the LTS driver. 30898 is relatively new.
So no, we can't just remove it, at least not yet.

Reply:
We have an internal task to look into this in the future. At the moment it's not possible because it's still an experimental feature. It's also not clear if the implementation of zeCommandListImmediateAppendCommandListsExp has been properly tested on older GPU's and we still need to support those for Graphs.

Reply:
Agree having 2 paths is definitely a maintenance burden, and I would very much like us to only have the new path in future. As has been said, in the short term we are blocked by:
1. LTS/public driver being recent enough to support this L0 feature
2. Verification on older GPUs for correctness/performance regressions.